src/components/NewBooks.js

```diff
@@ -0,0 +1,67 @@
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { addBook, removeBook } from '../redux/books/booksSlice';
+
+const BookActions = () => {
+  const dispatch = useDispatch();
+  const [newBook, setNewBook] = useState({
+    id: '',
+    title: '',
+    author: '',
+    category: '',
+  });
+
+  const handleAddBook = () => {
+    dispatch(addBook(newBook));
+    setNewBook({
+      id: '',
+      title: '',
+      author: '',
+      category: '',
+    });
+  };
+
```

Comment:
- [ ] I'm not able to add a book, It's giving the below error on the console. Please make sure to fix this issue and users should be able to add books without any error.

![image](https://github.com/Umuhan/Bookstore/assets/83803414/321c63fb-d6bc-4ed3-837c-f5085b313778)

Reply:
Thanks, solved